test/rekt/features/apiserversource/readiness.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserversource
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+	v1 "knative.dev/eventing/pkg/apis/sources/v1"
+	"knative.dev/eventing/test/rekt/resources/account_role"
+	"knative.dev/eventing/test/rekt/resources/apiserversource"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/manifest"
+	"knative.dev/reconciler-test/resources/svc"
+)
+
+// GoesReady returns a feature creating an ApiServerSource and then testing if an ApiServerSource becomes ready.
+func GoesReady(name string, cfg ...manifest.CfgFn) *feature.Feature {
+	f := feature.NewFeatureNamed("ApiServerSource goes ready.")
+
+	sink := feature.MakeRandomK8sName("sink")
+	f.Setup("install a service", svc.Install(sink, "app", "rekt"))
+
+	sacmName := feature.MakeRandomK8sName("apiserversource")
+	f.Setup("Create Service Account for ApiServerSource",
+		account_role.Install(sacmName,
```

Comment:
If you look at the broker/channel changes we found that this pattern of combining `Install` with `GoesReady` was not idea for test composition (my bad). I would recommend this being split into two methods: `GoesReady` which tests that the thing you pass is does go ready (and we have some helpers for that. and `Install` which installs the ApiServerSource resource.